src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseStartsWithInsteadOfIndexOfComparisonWithZero.cs

```diff
@@ -0,0 +1,128 @@
+                AddIfNotNull((indexOf.SingleOrDefault(s => s.Parameters is [{ Type.SpecialType: SpecialType.System_String }, { Name: "comparisonType" }]), OverloadString_StringComparison));
+                AddIfNotNull((indexOf.SingleOrDefault(s => s.Parameters is [{ Type.SpecialType: SpecialType.System_Char }, { Name: "comparisonType" }]), OverloadChar_StringComparison));
+                if (indexOfMethodsBuilder.Count == 0)
+                {
+                    return;
+                }
+
+                var indexOfMethods = indexOfMethodsBuilder.ToImmutable();
+
+                context.RegisterOperationAction(context =>
+                {
+                    var binaryOperation = (IBinaryOperation)context.Operation;
+                    if (binaryOperation.OperatorKind is not (BinaryOperatorKind.Equals or BinaryOperatorKind.NotEquals))
+                    {
+                        return;
+                    }
+
+                    if (IsIndexOfComparedWithZero(binaryOperation.LeftOperand, binaryOperation.RightOperand, indexOfMethods, out var additionalLocations, out var properties) ||
+                        IsIndexOfComparedWithZero(binaryOperation.RightOperand, binaryOperation.LeftOperand, indexOfMethods, out additionalLocations, out properties))
+                    {
+                        if (binaryOperation.OperatorKind == BinaryOperatorKind.NotEquals)
+                        {
+                            properties = properties.Add(ShouldNegateKey, "");
+                        }
+
+                        context.ReportDiagnostic(binaryOperation.CreateDiagnostic(Rule, additionalLocations, properties));
+                    }
+
+                }, OperationKind.Binary);
+
+                void AddIfNotNull((IMethodSymbol? Symbol, string OverloadPropertyValue) item)
+                {
+                    if (item.Symbol is not null)
+                    {
+                        indexOfMethodsBuilder.Add((item.Symbol, item.OverloadPropertyValue));
+                    }
+                }
+            });
+        }
+
+        private static bool IsIndexOfComparedWithZero(
+            IOperation left, IOperation right,
+            ImmutableArray<(IMethodSymbol Symbol, string OverloadPropertyValue)> indexOfMethods,
+            out ImmutableArray<Location> additionalLocations,
+            out ImmutableDictionary<string, string?> properties)
+        {
+            properties = ImmutableDictionary<string, string?>.Empty;
+
+            if (right.ConstantValue is { HasValue: true, Value: 0 } &&
+                left is IInvocationOperation invocation)
+            {
+                foreach (var (indexOfMethod, overloadPropertyValue) in indexOfMethods)
+                {
+                    if (indexOfMethod.Parameters.Length == invocation.Arguments.Length && indexOfMethod.Equals(invocation.TargetMethod, SymbolEqualityComparer.Default))
```

Comment:
NIT: Target method and the comparing method symbols will not be equal if the parameters/arguments are different, no? Feels like `indexOfMethod.Parameters.Length == invocation.Arguments.Length` comparison is redundant